### PR TITLE
Add pyux test

### PR DESCRIPTION
**Pull request by farizrahman4u**
#12073 clone

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    exceptions = [
+        pyux.ADDED_ARG_WITH_DEFAULT_IN_METHOD,
+        pyux.ADDED_DEFAULT_IN_METHOD
+    ]
+
+    diff = list(filter(lambda c: c[0] not in exceptions, diff))
+
+    if diff:
+        str_diff = '\n'.join([str(x) for x in diff])
+        raise Exception("API change detected ! \n " + str_diff)
```
Comment (fchollet) [inline tests/test_api.py]: Could we get an item-by-item print of all differences here? Comparing e.g. the old signature to the new signature for each symbol that has changed.

Comment (farizrahman4u) [reply tests/test_api.py]: `diff` is a list of tuples, each tuple of form `(Type of change, Item which changed, before value, after value)`. Would you like to see some sample diffs?

File: tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import keras.utils.test_utils
+
+
+def test_api():
+    api_file = os.path.join(os.getcwd(), 'api.json')
+
+    with open(api_file, 'r') as f:
+        previous_api = json.load(f)
+
+    current_api = pyux.sign(keras)
```
Comment (fchollet) [inline tests/test_api.py]: Style nit: there's a lot of whitespace in this function, you can group semantically related lines into paragraph.

File: tests/test_api.py (hunk 1 above)
Comment (fchollet) [reply tests/test_api.py]: Sure, seeing some sample diffs would be great!

Btw is there a more meaningful exception type we could raise instead of `Exception`?

**Comment by gabrieldemarmiesse** [comment]
As a note: I noticed that PyGithub https://github.com/PyGithub/PyGithub provides a very nice api to interact with github. I played a bit with it and I see that it shouldn't be hard to include it in travis :) Once this PR is merged, I'll try to make it work with pyux in travis.

The best workflow would be:
1) travis detects an API change
2) travis apply a label "API changes need appoval" to the corresponding pull request and comment to tell us what have changed in the API.
3) @fchollet can review the API changes and change the label to "API changes approved"
4) @fchollet or another maintainer can review the rest of the changes which are not related to the API. 

We should split the API review and code review to avoid putting too many PRs on @fchollet 's todo list.

File: tests/test_api.py (hunk 1 above)
Comment (farizrahman4u) [reply tests/test_api.py]: Resolved - we use `pyux.APIChangedException` instead.

File: tests/test_api.py (hunk 2 above)
Comment (farizrahman4u) [reply tests/test_api.py]: done

**Comment by farizrahman4u** [comment]
@gabrieldemarmiesse  @fchollet Lets discuss Github integrations?

**Comment by gabrieldemarmiesse** [comment]
So the thing is that to add tags to the pull requests, we need to
authenticate using a private API key from github. The issue here is
security. We cannot allow people doing pull requests from their fork from
manipulating this API key with arbitrary code (they can easily display it).
So we can only use this API key in the master branch build or in a cron job
which uses the master branch.

To solve this, I propose that we run the cron job with travis from the
master branch of another repo. Something like 'keras-bot'. It doesn't seem
too hard to do. What do you think about it?

**Comment by gabrieldemarmiesse** [comment]
@farizrahman4u I made a bot for keras-contrib, see https://github.com/keras-team/keras-contrib/issues/416 Would you like me to use this approach for the pyux?

**Comment by farizrahman4u** [comment]
Yes, please. I guess that also solves the issue with security?

**Comment by gabrieldemarmiesse** [comment]
Yes indeed. A long as we control what goes in the master branch.

This is not hard to do, but I have quite a lot of things on my plate, so it might take a while.
